src/components/typeahead/typeahead.vue

```diff
@@ -0,0 +1,259 @@
+                }
+                if (this.highlightedIndex > this.filteredItems.length - 1) {
+                    this.highlightedIndex = this.filteredItems.length - 1
+                }
+            },
+
+            defaultResultSlotMarkup(item, firstOnly = false) {
+                let matches = []
+                this.filterKeys.some(key => {
+                    const stringKey = String(item[key]).toLowerCase();
+                    if (stringKey.indexOf(this.query.toLowerCase()) > -1) {
+                        matches.push(`${key}: ${item[key]}`)
+                        if (firstOnly) {
+                            return true
+                        }
+                    }
+                })
+                return matches.join('. ');
+            }
+        },
+
+        computed: {
+            isEmpty() {
+                if( typeof this.filteredItems === 'undefined'  ) {
+                    return false
+                } else {
+                    return this.filteredItems.length < 1
+                }
+
+            },
+        },
+    }
+</script>
+<template lang="pug">
+    .pd-typeahed
+        .border.border-light.p-1.d-flex.flex-wrap
+            .result-block.bg-primary.text-white.py-1.px-3.m-1.d-flex.align-items-center(v-for="(item, key) in results", :key="key")
+                .label(v-text="$scopedSlots['result-text'] ? $scopedSlots['result-text']({item})[0].text : item[internalKeys.resultLabel]")
+                .delete(@click="removeFromResults(key)") ×
+            input#tpInput.border-0.ml-1(
+                ref="input",
+                type="text",
+                @input="onInput",
+                @focus="onFocus",
+                @blur="onBlur",
+                @keydown="onKeydown",
+                :placeholder="placeholder",
+                v-model="query"
+            )
+        section(v-if="focused")
+            ul.list-unstyled.position-absolute.results-list
+                li.border.border-top-white(
+                v-for="(item, key) in filteredItems",
+                :key="key",
+                @mousedown="selectItem(item)",
+                :class="{'hightlighted': highlightedIndex == key}")
+                    slot(name="result-list", :item="item")
+                        .p-1(v-text="defaultResultSlotMarkup(item)")
+            .p-1(v-show="isEmpty")
+                slot(name="no-results") No results matched your query.
+
+</template>
+<style scoped>
+    .results-list {
+        z-index: 1;
+        background: white;
+        max-height: 400px;
+        overflow-y: scroll;
+    }
+    .result-block {
+        z-index: 0;
+    }
+    .result-block .delete{
+        font-size: 16px;
+        margin-left: 10px;
+        cursor: pointer;
+    }
+    #tpInput {
+        outline: 0;
+        flex: 1;
+    }
+    .results-list > li {
+        cursor: pointer;
+    }
+    .results-list > li.hightlighted {
+        border: 1px solid #00adef !important;
```

Comment:
This should be outline;
Also can't hardcode color

Reply:
Agreed.
How do i un-hardcode color?